### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by kifle23**
The following features are added to this PR:

- [x] Created class classroom
- [x] Updated Classroom and Student with a has-many/belongs-to relationship
- [x] Created class book
- [x] Created class rental
- [x] Created person-book many-to-many relationships via rental class

### CODE REVIEW

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# The Book class represents a book with a title and an author.
+# It provides getters and setters for the title and author.
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
```
Comment (Davidosky007) [inline lib/book.rb:8]: * Kindly ensure `A method for adding rentals be present` which shows the relationship `* A book has many rentals` [Check requirements](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20has%2Dmany%20side%20of%20Book%20and%20Rental%20(a%20book%20has%20many%20rentals).) 
* Link to the example in a previous lesson: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship

File: lib/student.rb [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
-  attr_reader :classroom
+  attr_accessor :classroom
 
-  def initialize(name = 'Unknown', age = 0, parent_permission: true, classroom: 'Unknown')
-    super(name, age, parent_permission)
-    @classroom = classroom
+  def initialize(name = 'Unknown', age = 0, parent_permission: true, classroom: nil)
+    super(name, age, parent_permission: parent_permission)
+    @classroom = nil
+    self.classroom = classroom if classroom
```
Comment (Davidosky007) [inline lib/student.rb:10]: * Kindly add ` a method that serves as the 'belongs-to' side (a student belongs to a classroom)` 
* Link to the article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships
* [Check requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20belongs%2Dto%20side%20(a%20student%20belongs%20to%20a%20classroom).)

File: lib/person.rb [hunk 3]
```diff
@@ -3,14 +3,15 @@
...
+  attr_accessor :name, :age, :parent_permission, :rentals
 
   def initialize(name = 'Unknown', age = 0, parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (Davidosky007) [inline lib/person.rb:16]: * Kindly ensure `A method for adding rentals  be present` which shows the relationship `* A person has many rentals`
* Link to the example in a previous lesson: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship
* [Check requirement](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_associations.md#:~:text=Create%20the%20has%2Dmany%20side%20of%20Person%20and%20Rental%20(a%20person%20has%20many%20rentals).)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @kifle23 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* Ruby syntax written properly 👍 
* Well written PR 👍 
* Good looking readme 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/person.rb (hunk 3 above)
Comment (kifle23) [reply lib/person.rb:16]: Thank you so much @Davidosky007

File: lib/book.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (danmainah) [inline lib/book.rb]: - [ ] @kifle23 great work adding a method to add rental in the book class, however in order to achieve the has-many relationship the method needs to include person, date and the book itself.
```suggestion
  def add_rental(person, date)
     Rental.new(date, self, person)
  end
```

File: lib/person.rb [hunk 5]
```diff
@@ -29,5 +30,9 @@ def to_s
     "Person (id: #{@id}, name: #{@name}, age: #{@age}, parent_permission: #{@parent_permission})"
   end
 
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (danmainah) [inline lib/person.rb]: - [ ]  @kifle23 Great work adding a method to add rental in the person class, however in order to achieve the has-many relationship the method needs to include book, date, and the person himself. (kindly check the suggested code in the book class)

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @kifle23 , 

Great work on making the changes requested by a previous reviewer 👏  👏 👏
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
  ✔️ Implemented changes requested to add add rental methods


 # Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/person.rb (hunk 5 above)
Comment (kifle23) [reply lib/person.rb]: Thank you so much @danmainah

**Review by nahnahdev — APPROVED** [review]
## Project Status: APPROVED :green_circle: 
Hi @kifle23  , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :fire: 
- All requested changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by kifle23** [comment]
Thank you so much @NahnahAJ

**Pull request merged**